File: 7kyu 2/SeriesSum.js
/*
|--------------------------------------------------------------------------
| Sum of the first nth term of Series
|--------------------------------------------------------------------------
|
*/

function SeriesSum(n) {
   if (n == 0) {
      return "0.00";
   } else {
      let res = 1;
      let x = 4;
      for (let i = 1; i < n; i++) {
         let ser = 1 / x;
         res += ser;
         x += 3;
      }
      return res.toFixed(2);
   }
}


function SeriesSum(n, s = 0) {
   return n ? SeriesSum(n - 1, s + 1 / (3 * n - 2)) : s.toFixed(2)
}


function SeriesSum(n) {
   for (let s = 0, i = 0; i < n; i++) {
      s += 1 / (1 + i * 3)
   }
   return s.toFixed(2)
}


function SeriesSum(n) {
   for (a = 0, i = 1; i <= n * 3; i += 3) a += 1 / i;
   return a.toFixed(2);
}


function SeriesSum(n) {
   let s = 0;
   while (n) {
      s += 1 / (1 + 3 * --n);
   }
   return s.toFixed(2);
}


console.log(SeriesSum(1), "1.00")
console.log(SeriesSum(2), "1.25")
console.log(SeriesSum(3), "1.39")
console.log(SeriesSum(4), "1.49")
console.log(SeriesSum(0), "0.00")

File: Python/7py 2/series_sum.py
"""
|--------------------------------------------------------------------------
| Sum of the first nth term of Series
|--------------------------------------------------------------------------
|
"""

def series_sum(n):   
   if n == 0:
      return "0.00"
   else:
      res = 1
      x = 4
      for i in range(1, n):
         ser = 1 / x
         res += ser
         x += 3
      return format(res, '.2f')


def series_sum(n):
   return '{:.2f}'.format(sum(1.0/(3 * i + 1) for i in range(n)))


def series_sum(n):
   sum = 0.0
   for i in range(0,n):
      sum += 1 / (1 + 3 * float(i))
   return '%.2f' % sum


def series_sum(n):
   return f'{sum(1/d for d in range(1,n*3,3)):.2f}'


print(series_sum(1), "1.00")
print(series_sum(2), "1.25")
print(series_sum(3), "1.39")